File: xml-data/src/main/java/artofillusion/plugin/History.java
package artofillusion.plugin;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamImplicit;


import java.util.ArrayList;
import java.util.List;

@XStreamAlias("history")
public class History {

    @XStreamImplicit
    private List<LogRecord> records = new ArrayList<>();

    public List<LogRecord> getRecords() {
        return records == null ? List.of() : records;
    }

}

File: xml-data/src/main/java/artofillusion/plugin/History.kt
package artofillusion.plugin

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamImplicit
import java.util.ArrayList

@XStreamAlias("history")
class History {
    @XStreamImplicit
    private val records: MutableList<LogRecord?>? = ArrayList<LogRecord?>()

    fun getRecords(): MutableList<LogRecord?> {
        return (if (records == null) kotlin.collections.mutableListOf<artofillusion.plugin.LogRecord?>() else records)
    }
}
